pyteal/ast/gtxn.py

```diff
@@ -47,10 +51,21 @@ def __teal__(self, options: "CompileOptions"):
 class GtxnaExpr(TxnaExpr):
     """An expression that accesses a transaction array field from a transaction in the current group."""
 
+    @staticmethod
+    def __validate_txn_index_or_throw(txnIndex: Union[int, Expr]):
+        is_expr = isinstance(txnIndex, Expr)
+        if not (type(txnIndex) is int or is_expr):
+            raise TealInputError(
+                f"Invalid txnIndex type:  Expected int or Expr, but received {txnIndex}"
+            )
+        if is_expr:
+            require_type(txnIndex, TealType.uint64)
+
     def __init__(
         self, txnIndex: Union[int, Expr], field: TxnField, index: Union[int, Expr]
     ) -> None:
         super().__init__(Op.gtxna, Op.gtxnas, "Gtxna", field, index)
+        self.__validate_txn_index_or_throw(txnIndex)
```

Comment:
@jasonpaulos Sorry - errant mistake on my part.  I fixed the type checking via https://github.com/michaeldiamant/pyteal/pull/1/commits/4b3440a06e5e9acc553679664ea486ed07b579b6.

Aside:  As a tertiary concern, I'd be interested in understanding the Pythonic approach to share the validation function between 2 classes.  Since the duplication doesn't feel like a pressing concern to me, I opted to present the PR without further research.

Reply:
For the sake of getting this PR merged, I don't mind the duplication.

But in the spirit of sharing best practices, I'd suggest making the validation function a top-level function instead of being inside one/both of the classes.